### PR TITLE
Enabled hooks, defined task switch hook

### PR DESCRIPTION
**Pull request by Cam0Cow**
#323

### CODE REVIEW

**Comment by Cam0Cow** [comment]
Tested on hardware, seems to work just fine

File: Apps/Inc/Tasks.h [hunk 1]
```diff
@@ -152,6 +158,20 @@ typedef enum{
...
+/**
+ * Task trace
+ * 
+ * Stores the last 8 tasks that were run
+ * The most recent task is at tasks[index], the one before at tasks[index-1],
+ * wrapping back around at the beginnning
+ */
+typedef struct {
+    OS_TCB *tasks[8];
+    uint32_t index;
```
Comment (NathanielDelgado) [inline Apps/Inc/Tasks.h:170]: Doesn't matter but change to uint8_t to save space.

File: Apps/Inc/Tasks.h [hunk 2]
```diff
@@ -152,6 +158,20 @@ typedef enum{
...
 
+/**
+ * Task trace
+ * 
+ * Stores the last 8 tasks that were run
+ * The most recent task is at tasks[index], the one before at tasks[index-1],
+ * wrapping back around at the beginnning
+ */
+typedef struct {
+    OS_TCB *tasks[8];
```
Comment (NathanielDelgado) [inline Apps/Inc/Tasks.h:169]: Maybe make the 8 a macro?

**Review by NathanielDelgado — APPROVED** [review]
Looks good

File: Apps/Inc/Tasks.h (hunk 1 above)
Comment (Cam0Cow) [reply Apps/Inc/Tasks.h:170]: Due to padding it actually wouldn't make a difference.

File: Apps/Inc/Tasks.h (hunk 2 above)
Comment (Cam0Cow) [reply Apps/Inc/Tasks.h:169]: Good idea, I'll do that

**Review by IshDeshpa — APPROVED** [review]
Looks good. It may be worth noting that we could just store previous task if we enter a fault state or some built-in fault handler, but assuming that this does not add much overhead to our context switches, this is cool.

**Review by NathanielDelgado — APPROVED** [review]
✔

**Pull request merged**